.github/workflows/ci.yml

```diff
@@ -69,11 +110,6 @@ jobs:
         run: |
           make lint-and-type-check
 
-      # TODO: vendor only the parts that we need from the following action
```

Comment:
Goodbye to douglascamata, whoever he is? :-)

Do we actually/still test on MacOS?

Reply:
This made me chuckle. :laughing: 

**TL;DR** Your work with the mock, and the issues you have opened on CI brittleness, have convinced me that we should remove Docker from our CI. An immediate win from this is that we can run our CI on all major platforms without any extra work (like the douglascamata action above that is specific to MacOS).

**Longer Answer**
At the moment, we only test on Linux - I couldn't figure out how to get our docker setup to work on Windows, and it was taking an age (>5 minutes every time) to set it up on Mac. This action from douglascamata was required to run on Mac, and so I kept it as I thought that some day we would have the Docker working in the CI for all major platforms.

But with your work on the `pymongo` mock, and with all the issues you have flagged with our CI failing so regularly with Docker, I think a far easier solution is to phase out Docker from our core CI action (ie the action that runs on every PR). This way, we don't need to do anything special to support Windows or MacOS on CI, a very nice 'win'!